### PR TITLE
FLUID-6127: Fix and test cases for faulty mapping of "DELETE" changes by wildcard modelListeners

### PR DESCRIPTION
**Pull request by amb26**

### CODE REVIEW

File: tests/framework-tests/core/js/DataBindingTests.js [hunk 1]
```diff
@@ -911,6 +911,97 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
...
+        model: {
+            idToPath: {
+                first: 1
+            }
+        }
+    });
+
+    fluid.tests.fluid6127fixtures = [{
+        change: {
+            type: "ADD",
```
Comment (cindyli) [inline tests/framework-tests/core/js/DataBindingTests.js]: Test cases here cover change type `ADD` and `DELETE`. Shall test cases when type is not specified (the `modify` case) be added?

Comment (amb26) [inline tests/framework-tests/core/js/DataBindingTests.js]: There is currently no other separate change type in the system - the default when the type is not specified is "ADD"

Comment (cindyli) [inline tests/framework-tests/core/js/DataBindingTests.js]: OK, I see.

**Comment by cindyli** [comment]
Merged at 00f931479c38605f16106e474695e9eb624748e5

**Pull request merged**